tests/src/test/scala/common/TestUtils.java

```diff
@@ -327,6 +327,7 @@ public String call() throws IOException {
             logger.info("RunResult: " + rr);
         }
         rr.validateExitCode(expectedExitCode);
+        executor.shutdown();
```

Comment:
I'm tempted to suggested making the executor a static instance. What do you think?